hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/schema/TableResolverImpl.java

```diff
@@ -166,14 +169,29 @@ public void removeView(String name, boolean ifExists) {
     @Nonnull
     @Override
     public List<Table> getTables() {
-        Collection<Mapping> mappings = tableStorage.valuesMappings();
-        List<Table> tables = new ArrayList<>(mappings.size() + 3);
-        for (Mapping mapping : mappings) {
-            tables.add(toTable(mapping));
+        Collection<Object> objects = tableStorage.allObjects();
+        List<Table> tables = new ArrayList<>(objects.size() + 3);
+        for (Object o : objects) {
+            if (o instanceof Mapping) {
+                tables.add(toTable((Mapping) o));
+            } else if (o instanceof View) {
+                tables.add(toTable((View) o));
+            } else {
+                throw new RuntimeException("Unexpected: " + o);
+            }
         }
+
+        Collection<Mapping> mappings = objects.stream()
+                .filter(o -> o instanceof Mapping)
+                .map(m -> (Mapping) m)
+                .collect(Collectors.toList());
+        Collection<View> views = objects.stream()
```

Comment:
This is a good candidate for introducing util method for avoiding a repeating code. Something like:
```
   private static <T>Collection<T> get(Collection<Object> objects, Class<T> clazz) {
        return objects.stream().filter(clazz::isInstance).map((clazz::cast)).collect(Collectors.toList());
    }
```

Reply:
I did this line. I was thinking about this, but didn't do it because i didn't want to use reflection...

Reply:
I wouldn't call it using reflection. We are not getting class information like fields, methods, etc. Those two methods from `Class` class are intrinsics most likely.

Reply:
well, i don't mind...